frontend/mobile/lib/pages/library.dart

```diff
@@ -32,34 +35,77 @@ class _LibraryState extends State<Library> {
-                        });
-                      },
-                    );
+          child: Column(
+            children: [
+              Padding(
+                padding: const EdgeInsets.all(4.0),
+                child: Container(
+                  color: Colors.white,
+                  child: TextField(
+                    controller: searchController,
+                    decoration: InputDecoration(
+                      hintText: AppLocalizations.of(context)!.searchBar,
+                      suffixIcon: IconButton(
+                        icon: const Icon(Icons.clear),
+                        onPressed: () => searchController.clear(),
+                      ),
+                      prefixIcon: IconButton(
+                        icon: const Icon(Icons.search),
+                        onPressed: () async {
+                          setState(() {
+                            workflows = initialWorkflows
+                                .where((element) => element.name
+                                    .contains(searchController.text))
+                                .toList();
+                          });
+                        },
+                      ),
+                      border: OutlineInputBorder(
+                        borderRadius: BorderRadius.circular(0.0),
+                      ),
+                    ),
+                  ),
+                ),
+              ),
+              Expanded(
+                child: FutureBuilder<ServiceReturn<List<Workflow>>>(
+                  future: services.workflows.getAll(),
+                  builder: (context, snapshot) {
+                    if (snapshot.connectionState == ConnectionState.waiting) {
+                      return const Center(child: CircularProgressIndicator());
+                    } else if (snapshot.hasError) {
+                      return Center(
+                        child: Text(AppLocalizations.of(context)!
+                            .error(snapshot.error.toString())),
+                      );
+                    } else {
+                      if (needRefresh) {
+                        needRefresh = false;
+                        if (snapshot.data!.data != null) {
+                          initialWorkflows = snapshot.data!.data!;
+                          workflows = snapshot.data!.data!;
+                        }
+                      }
+                      return ListView.builder(
+                        itemCount: workflows.length,
+                        itemBuilder: (context, index) {
+                          return WorkflowTile(
+                            workflow: workflows[index],
+                            onUpdate: () async {
+                              final newWorkflows =
+                                  await services.workflows.getAll();
+                              setState(() {
+                                workflows = newWorkflows.data!;
```

Comment:
apply the search here too, or when somebody renames a workflow it won't be applied

Reply:
or you can also clear the search, but the first solution is better imo

Reply:
this is only ran once at the page loading, so no need to apply the search here

Reply:
it's also ran when somebody updates the workflow (rename or delete for example).
So here I have that if I have 5 workflows, search for "unknown" (that marches a single workflow), if I delete or rename it then all the other workflows will appear again

Reply:
fixed